godot-core/src/builtin/plane.rs

```diff
@@ -191,6 +313,281 @@ mod test {
+        // From a point straight down the origin point, a segment pointing straight up would cross the `plane`
+        // in the origin point only if the segment ended on or beyond the `plane`.
+        assert_eq!(plane.intersects_segment(vec_d, vec_a), Some(vec_a));
+        assert_eq!(plane.intersects_segment(vec_d, vec_b), Some(vec_a));
+        assert_eq!(plane.intersects_segment(vec_d, vec_c), Some(vec_a));
+        assert_eq!(plane.intersects_segment(vec_d, vec_e), None);
+        // From a point straight up the origin point, the distance from the origin to the intersection point
+        // formed by a slanted segment should follow the Pythagorean Theorem only if the segment ended on or 
+        // beyond the `plane`.
+        assert_eq!(plane.intersects_segment(vec_c, vec_f - vec_c).unwrap().length(), 
+                   (vec_c.distance_squared_to(vec_f) - vec_c.length_squared()).sqrt());
+        assert_eq!(plane.intersects_segment(vec_b, vec_f - vec_b).unwrap().length(), 
+                   (vec_b.distance_squared_to(vec_f) - vec_b.length_squared()).sqrt());
+        // A segment parallel to the `plane` would not intersect the plane.
+        assert_eq!(plane.intersects_segment(vec_c, vec_g), None);
+        // A segment pointing to the opposite direction as the `plane` would not intersect it.
+        assert_eq!(plane.intersects_segment(vec_b, vec_c), None);
+        assert_eq!(plane.intersects_segment(vec_b, vec_f), None);
+    }
+
+    /// Tests `is_equal_approx()`.
+    #[test]
+    fn test_equal() {
+        // Initial normalized planes.
+        let mut plane_a: Plane = Plane::new(Vector3::new(0.0, 0.0, 1.0).normalized(), 0.0);
+        let mut plane_b: Plane = Plane::new(Vector3::new(0.01, 0.0, 1.0).normalized(), 0.0);
+        let mut plane_c: Plane = Plane::new(Vector3::new(0.0001, 0.0, 1.0).normalized(), 0.0);
+        let mut plane_d: Plane = Plane::new(Vector3::new(0.000001, 0.0, 1.0).normalized(), 0.0);
+        let mut plane_e: Plane = Plane::new(Vector3::new(0.000001, 0.0, 1.0).normalized(), 0.01);
+        let mut plane_f: Plane = Plane::new(Vector3::new(0.000001, 0.0, 1.0).normalized(), 0.000001);
+
+        // Same planes should be equals.
+        assert!(plane_a.is_equal_approx(&plane_a));
+        assert!(plane_b.is_equal_approx(&plane_b));
+        // Although similar, planes below shouldn't be approximately equals.
+        assert_eq!(plane_a.is_equal_approx(&plane_b), false);
+        assert_eq!(plane_a.is_equal_approx(&plane_c), false);
+        // Planes below should be approximately equal because it's lower than the set tolerance constant.
+        assert!(plane_a.is_equal_approx(&plane_d));
+        // Although approximately equal in the `normal` part, it is not approximately equal in the `d` 
+        // part.
+        assert_eq!(plane_a.is_equal_approx(&plane_e), false);
+        // Both attributes are approximately equal.
+        assert!(plane_a.is_equal_approx(&plane_f));
```

Comment:
you should be able to use `assert_eq_approx`/`assert_ne_approx` here something like
```suggestion
        assert_ne_approx!(&plane_a, &plane_e, Plane::is_equal_approx);
        // Both attributes are approximately equal.
        assert_eq_approx!(&plane_a, &plane_f, Plane::is_equal_approx);
```